CMakeLists.txt

```diff
@@ -125,12 +125,14 @@ ENDIF(BUILD_WITH_CASADI_SUPPORT)
 
 SET(BOOST_REQUIRED_COMPONENTS filesystem serialization system)
 SET(BOOST_BUILD_COMPONENTS unit_test_framework)
-SET(BOOST_OPTIONAL_COMPONENTS "")
+
+ADD_PROJECT_DEPENDENCY(Boost COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
+FIND_PACKAGE(Boost COMPONENTS ${BOOST_BUILD_COMPONENTS})
```

Comment:
Shouldn't there be a *REQUIRED* somewhere here ?
I never know whether it is before or after *COMPONENTS*. I think @nim65s can shade light on this.

Reply:
This is now done. Thanks for the comment.